android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -1314,17 +1314,33 @@ private void setBackgroundImage(NTPImage ntpImage) {
         ViewTreeObserver observer = mBgImageView.getViewTreeObserver();
         observer.addOnGlobalLayoutListener(
                 new ViewTreeObserver.OnGlobalLayoutListener() {
+                    private int mLastWidth;
+                    private int mLastHeight;
+
                     @Override
                     public void onGlobalLayout() {
-                        mWorkerTask =
-                                new FetchWallpaperWorkerTask(
-                                        ntpImage,
-                                        mBgImageView.getMeasuredWidth(),
-                                        mBgImageView.getMeasuredHeight(),
-                                        mWallpaperRetrievedCallback);
-                        mWorkerTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
-
-                        mBgImageView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
```

Comment:
We still need to remove the listener, it could potentially causing memory leaks if not removed. I would do it inside `onDetachedFromWindow`. Create a member `private ViewTreeObserver.OnGlobalLayoutListener mBackgroundImageLayoutListener;` and clean it up
```
        if (mBackgroundImageLayoutListener != null && mBgImageView != null) {
            mBgImageView.getViewTreeObserver().removeOnGlobalLayoutListener(
                    mBackgroundImageLayoutListener);
            mBackgroundImageLayoutListener = null;
        }
```